ols/src/query_helpers/docs_summarizer.py

```diff
@@ -127,56 +157,41 @@ def summarize(
                     service_context=service_context,
                     verbose=verbose,
                 )
-                logger.info(f"{conversation_id} Setting up query engine")
-                query_engine = index.as_query_engine(
-                    text_qa_template=summarization_template,
-                    verbose=verbose,
-                    streaming=False,
-                    similarity_top_k=1,
-                    node_postprocessors=[_NoLLMMetadataPostprocessor()],
-                )
 
-                logger.info(f"{conversation_id} Submitting summarization query")
-                summary = query_engine.query(query)
+                logger.info(f"{conversation_id}: Getting index data.")
+                rag_context_data = self._get_rag_data(index, query)
 
-                for source_node in summary.source_nodes:
-                    referenced_documents.append(
-                        self._file_path_to_doc_url(
-                            source_node.node.metadata["file_path"]
-                        )
-                    )
             except Exception as err:
                 logger.error(f"Error loading vector index: {err}")
-                use_llm_without_reference_content = True
         else:
             logger.info("Reference content is not configured")
-            use_llm_without_reference_content = True
 
-        if use_llm_without_reference_content:
+        rag_context, referenced_documents = self._format_rag_data(rag_context_data)
+
+        chat_engine = LLMChain(
+            llm=bare_llm,
+            prompt=CHAT_PROMPT,
+            verbose=verbose,
+        )
+        summary = chat_engine.invoke({"context": rag_context, "query": query})
```

Comment:
it seems worthy of note that you're using the same prompt template regardless of whether there actually is any context data to include or not.

while real installs will always have a rag db available, it's not hard to imagine that in the case of some questions all the returned RAG content is so low scoring that we don't want to include any of it. So should we have a separate prompt for cases where there is no rag results to inject?

Reply:
did you test this with both BAM and OpenAI providers?  i'm a little surprised the same code works for both, given the TODO statement/hack below that you're removing.

Reply:
> If content is not available then it will try to answer from the data it was trained on (if that is valid/relevant)..

I understand, but the prompt is going to say something like `using the information below, answer the user's question`, except there won't actually be any "information below" if we don't have a RAG db configured or there's no retrieved content we want to include.  So wouldn't it be better to have a different prompt when we are not adding retrieved context?